cloudformation/stack_topic_handler.js

```diff
@@ -41,162 +58,255 @@ async function handleBudgetAlert(event, context) {
+    StackName: stackName,
+  });
   const params = stackInfo.Stacks[0].Parameters;
   const newParams = [];
-  
+
   for (const p of params) {
     if (p.ParameterKey === "StackOffline") {
-        newParams.push({ ParameterKey: p.ParameterKey, ParameterValue: OFFLINE_SETTING });
+      newParams.push({
+        ParameterKey: p.ParameterKey,
+        ParameterValue: OFFLINE_SETTING,
+      });
     } else {
-        newParams.push({ ParameterKey: p.ParameterKey, UsePreviousValue: true });
+      newParams.push({ ParameterKey: p.ParameterKey, UsePreviousValue: true });
     }
   }
-  
+
   await promisify(cf.updateStack.bind(cf))({
-      StackName: stackName,
-      UsePreviousTemplate: true,
-      Parameters: newParams,
-      Capabilities: ["CAPABILITY_IAM", "CAPABILITY_NAMED_IAM"]
+    StackName: stackName,
+    UsePreviousTemplate: true,
+    Parameters: newParams,
+    Capabilities: ["CAPABILITY_IAM", "CAPABILITY_NAMED_IAM"],
   });
 }
 
 async function handleASGMessage(message, context) {
   const asgName = message.AutoScalingGroupName;
   const asgEvent = message.Event;
-  const region = "${AWS::Region}";
-  const recordName = "${LowerStackName.Value}-app.${InternalZoneInfo.Name}.";
-  const hostedZoneId = "${InternalZoneInfo.Id}";
+  const REGION = "${AWS::Region}";
+  const RECORD_NAME = "${LowerStackName.Value}-app.${InternalZoneInfo.Name}.";
+  const HOSTED_ZONE_ID = "${InternalZoneInfo.Id}";
   const ttl = 15;
+  const DOMAIN_URL = "${InternalZoneInfo.Name}.";
 
-  //console.log(asgEvent);
-
-  if (asgEvent === "autoscaling:EC2_INSTANCE_LAUNCH" || asgEvent === "autoscaling:EC2_INSTANCE_TERMINATE" || asgEvent === "INSTANCE_REBOOT") {
-    //console.log("Handling Launch/Terminate Event for " + asgName);
+  if (
+    asgEvent === "autoscaling:EC2_INSTANCE_LAUNCH" ||
+    asgEvent === "autoscaling:EC2_INSTANCE_TERMINATE" ||
+    asgEvent === "INSTANCE_REBOOT"
+  ) {
+    // console.log('Handling Launch/Terminate Event for ' + asgName)
 
-    const autoscaling = new AWS.AutoScaling({region});
-    const ec2 = new AWS.EC2({region});
+    const autoscaling = new AWS.AutoScaling({ region: REGION });
+    const ec2 = new AWS.EC2({ region: REGION });
     const route53 = new AWS.Route53();
 
-    const asgResponse = await promisify(autoscaling.describeAutoScalingGroups.bind(autoscaling))({
+    const asgResponse = await promisify(
+      autoscaling.describeAutoScalingGroups.bind(autoscaling)
+    )({
       AutoScalingGroupNames: [asgName],
-      MaxRecords: 1
+      MaxRecords: 1,
     });
 
-    const recordSets = (await promisify(route53.listResourceRecordSets.bind(route53))({
-      StartRecordName: recordName,
-      StartRecordType: "A",
-      HostedZoneId: hostedZoneId,
-      MaxItems: "100"
-    })).ResourceRecordSets;
+    const recordSets = (
+      await promisify(route53.listResourceRecordSets.bind(route53))({
+        HostedZoneId: HOSTED_ZONE_ID,
+        MaxItems: "100",
```

Comment:
I did, the reason being is it only gets the records that are "greater" than the specified StartRecordType. I attempted to use the string "." + DOMAIN_URL so only the records that look like subdomains would be included but the API threw an error.

Right now HC deployments (at least for mine) had >100 records because of the leftover A records, now only has 27 records like it's supposed to.

Reply:
I suppose the stale records past these 100 will eventually be flushed as instances roll over, or we could actually attempt to paginate though all records in this script, instead of stopping at 100.

Reply:
Since it had >100, this had to run twice to get them all. I opted to let people every time they scale up, down, or turn their instance to "offline mode" allow this to run and update the records only that much. 

Open to discussion on this one. I didn't want to implement another loop that could fail on this script, it feels fragile enough and I assume people when they set instances to "offline mode", scale up or down, would run this script enough times to finish the deletion.

Reply:
:+1: Gotcha, yeah, I agree with trying to keep this script as simple as possible. As long as there's a path towards getting all the records cleaned up, that sounds good.